src/Generic/Data/Internal/Enum.hs

```diff
@@ -52,7 +85,82 @@ gtoFiniteEnum n
 --
 -- See also 'gtoFiniteEnum'.
 gfromFiniteEnum :: (Generic a, GEnum FiniteEnum (Rep a)) => a -> Int
-gfromFiniteEnum = gFromEnum @FiniteEnum . from
+gfromFiniteEnum = gfromEnum' @FiniteEnum
+
+-- | Generic 'enumFrom' generated with the 'FiniteEnum' option.
+--
+-- See also 'gtoEnum'.
+gfiniteEnumFrom :: (Generic a, GEnum FiniteEnum (Rep a)) => a -> [a]
+gfiniteEnumFrom = genumFrom' @FiniteEnum
+
+-- | Generic 'enumFromThen' generated with the 'FiniteEnum' option.
+--
+-- See also 'gtoEnum'.
+gfiniteEnumFromThen :: (Generic a, GEnum FiniteEnum (Rep a)) => a -> a -> [a]
+gfiniteEnumFromThen = genumFromThen' @FiniteEnum
+
+-- | Generic 'enumFromTo' generated with the 'FiniteEnum' option.
+--
+-- See also 'gtoEnum'.
+gfiniteEnumFromTo :: (Generic a, GEnum FiniteEnum (Rep a)) => a -> a -> [a]
+gfiniteEnumFromTo = genumFromTo' @FiniteEnum
+
+-- | Generic 'enumFromThenTo' generated with the 'FiniteEnum' option.
+--
+-- See also 'gtoEnum'.
+gfiniteEnumFromThenTo :: (Generic a, GEnum FiniteEnum (Rep a)) => a -> a -> a -> [a]
+gfiniteEnumFromThenTo = genumFromThenTo' @FiniteEnum
+
+-- | Unsafe generic 'toEnum'. Does not check whether the argument is within
+-- valid bounds. Use 'gtoEnum' or 'gtoFiniteEnum' instead.
+gtoEnum' :: forall opts a. (Generic a, GEnum opts (Rep a)) => Int -> a
+gtoEnum' = to. gToEnum @opts
+
+-- | Generic 'fromEnum'. Use 'gfromEnum' or 'gfromFiniteEnum' instead
+gfromEnum' :: forall opts a. (Generic a, GEnum opts (Rep a)) => a -> Int
+gfromEnum' = gFromEnum @opts . from
+
+-- | > genumMin == gfromEnum gminBound
+genumMin :: Int
+genumMin = 0
+
+-- | > genumMin == gfromEnum gmaxBound
```

Comment:
Should be `genumMax` instead of `genumMin`.

Reply:
Thanks, fixed in https://github.com/Lysxia/generic-data/commit/ea753c3758c1b82f2a022fff1c954d8e362ecc43